Update benchmark dev doc (#8402)
knative/serving: test/performance/dev.md

Take `dataplane-probe` benchmark for example:

> 1. Edit [mako config](https://github.com/knative/serving/blob/master/test/performance/config/config-mako.yaml)
>    to attach your desired tags, see the `_example` stanza for how.
>    Then apply it:

Reviewer: <!--mdfmt-bot-->
Format markdown:
```suggestion
1. Edit
   [mako config](https://github.com/knative/serving/blob/master/test/performance/config/config-mako.yaml)
   to attach your desired tags, see the `_example` stanza for how. Then apply
   it:
```

   kubectl create secret generic mako-secrets --from-file=./robot.json
   ```

1. Patch Istio:

   ```shell

>    # This command will fail if no HPA for istio-ingressgateway exists.
>    # The failure can be ingored if the following command is executed.

Reviewer 2: ```suggestion
   # This command will fail if no HPA for istio-ingressgateway exists.
   # If that is the case, the error can be ignored.
```

Reviewer 2: Same below.
But LG otherwise.

Author: Done

1. Patch istio:

   ```shell
   kubectl patch deploy -n istio-system istio-ingressgateway \

>      --patch '{"spec": {"replicas": 10}}'

Reviewer 2: won't HPA override this value?

Author: If following https://github.com/knative/serving/blob/master/DEVELOPMENT.md, `istio-minimal.yaml` is installed. There's no HPA for instio-ingressgateay.

   kubectl patch hpa -n knative-serving activator --patch '{"spec": {"minReplicas": 10}}'
   ```

> 1. Apply `setup` for benchmark if there's any. Take `dataplane-probe` benchmark for example:

Reviewer: <!--mdfmt-bot-->
Format markdown:
```suggestion
1. Apply `setup` for benchmark if there's any. Take `dataplane-probe` benchmark
   for example:
```
